File: sdk/samples/FrameworkSamples/CLR/CompressionSample/vb/compression/zipconstants.vb
 '-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------
Imports System
Imports System.Windows.Forms


'/ <summary>
'/ This class contains ZipConstants used for gzip.
'/ </summary>

NotInheritable Class ZipConstants
    
    Private Sub New() 
    
    End Sub 'New
    Public Const DEFLATE As Byte = 4 'Indicating deflate method of compression
    Public Const GZIP As Byte = 6 'Indicating gzip method of compression
    Public Const FixedHeaderSize As Integer = 30 'Fixed part of the header
    Public Const SuperHeaderSize As Integer = 3
    
    Public Const AddFiles As String = "Add files"
    Public Const NewArchive As String = "New archive"
    Public Const OpenArchive As String = "Open archive"
    'Extensions for the files
    Public Const Extension As String = "xip files (*.xip)|*.xip"
    Public Const AllExtensions As String = "All files (*.*)|*.*"
    
    Public Const BackSlash As String = "\"
    Public Const Dot As String = "."
    
    Public Const ExtractMessage As String = "Extracted files to {0} successfully."
    Public Const OpenMessage As String = "Opened archive {0} for editing"
    Public Const NewMessage As String = "Opened new archive {0} for editing"
    Public Const AddMessage As String = "Added files to the archive."
    
    Public Const Title As String = "Compression sample"
    Public Const GzipName As String = "GZip"
    Public Const DeflateName As String = "Deflate"
    Public Const ErrorName As String = "Error"
    
    Public Const SeekError As String = "Attempted to seek before the beginning"
    Public Const IOError As String = "An IO error has occured"
    Public Const CloseError As String = "Stream already closed"
    Public Const MemoryError As String = "Running out of memory"
    Public Const ArgumentError As String = "Argument invalid"
    Public Const FileNotFoundError As String = "File not found"
    Public Const FileError As String = "Filename invalid"
    Public Const CorruptedError As String = "Trying to read a corrupted xip file"
    Public Const FileExistsError As String = "File already exists"
    
    Public Const FileReplace As String = "The file chosen already exists. Do you want to replace it?"
    Public Const Replace As String = "Replace"
    
    Public Shared Sub ShowError(ByVal [error] As String) 
        Dim opt As MessageBoxOptions
        If System.Threading.Thread.CurrentThread.CurrentUICulture.TextInfo.IsRightToLeft = True Then
            opt = MessageBoxOptions.RightAlign Or MessageBoxOptions.RtlReading
        Else
            opt = MessageBoxOptions.DefaultDesktopOnly
        End If
        MessageBox.Show([error], ErrorName, MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1, opt)
        CompressionForm.statusMessage = String.Empty
    
    End Sub 'ShowError
End Class 'ZipConstants

File: sdk/samples/FrameworkSamples/CLR/CompressionSample/cs/compression/zipconstants.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------
using System;
using System.Windows.Forms;

namespace Microsoft.Sample.Compression
{
	/// <summary>
	/// This class contains ZipConstants used for gzip.
	/// </summary>
	sealed class ZipConstants
	{
		private ZipConstants(){
		}
		public const byte DEFLATE = 4;	//Indicating deflate method of compression
		public const byte GZIP = 6;		//Indicating gzip method of compression

		public const int FixedHeaderSize = 30; //Fixed part of the header
		public const int SuperHeaderSize = 3;

		public const string AddFiles = "Add files";
		public const string NewArchive = "New archive";
		public const string OpenArchive = "Open archive";
		//Extensions for the files
		public const string Extension = "xip files (*.xip)|*.xip";
		public const string AllExtensions = "All files (*.*)|*.*";

		public const string BackSlash = "\\";
		public const string Dot = ".";

		public const string ExtractMessage = 
								"Extracted files to {0} successfully.";
		public const string OpenMessage = "Opened archive {0} for editing" ;
		public const string NewMessage = "Opened new archive {0} for editing";
		public const string AddMessage = "Added files to the archive.";

		public const string Title = "Compression sample";
		public const string GzipName = "GZip";
		public const string DeflateName = "Deflate";
		public const string ErrorName = "Error";

		public const string SeekError = "Attempted to seek before the beginning";
		public const string IOError = "An IO error has occured";
		public const string CloseError = "Stream already closed";
		public const string MemoryError = "Running out of memory";
		public const string ArgumentError = "Argument invalid";
		public const string FileNotFoundError = "File not found";
		public const string FileError = "Filename invalid";
		public const string CorruptedError = 
				"Trying to read a corrupted xip file";
		public const string FileExistsError = "File already exists";

		public const string FileReplace = 
			"The file chosen already exists. Do you want to replace it?";
		public const string Replace = "Replace";
		public static void ShowError(string error)
		{
			MessageBoxOptions opt;
			if (System.Threading.Thread.CurrentThread.
				CurrentUICulture.TextInfo.IsRightToLeft == true)
				opt = MessageBoxOptions.RightAlign | 
					MessageBoxOptions.RtlReading;
			else
				opt = MessageBoxOptions.DefaultDesktopOnly;
			MessageBox.Show(error, ErrorName, MessageBoxButtons.OK, 
				MessageBoxIcon.Error, MessageBoxDefaultButton.Button1, opt);
			CompressionForm.statusMessage = String.Empty;
		}
	}
}
